From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sat, 28 Dec 2013 21:25:06 -0500
Subject: [PATCH] Don't save Fireworks

Fireworks can bug out and not detonate, and an automated
launcher can very easily fill a chunk.

Prevent saving Fireworks so that chunk unloads will wipe a chunks fireworks in this case.
---
 src/main/java/net/minecraft/server/Entity.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index 5ab7615d6e714e331bedb65f5e1867c10d36eb91..7ff310ae615a63dcae7d92e860b3751e5e5f6f4b 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -1536,6 +1536,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
 
     public boolean serializeEntity(NBTTagCompound cmp) { List<Entity> pass = Lists.newArrayList(passengers); passengers.clear();boolean res = d(cmp); passengers.addAll(pass); return res; } // EMC // OBF HELPER
     public boolean d(NBTTagCompound nbttagcompound) {
+        if (this instanceof EntityFireworks) return false; // EMC
         return this.isPassenger() ? false : this.a_(nbttagcompound);
     }
 
